Patch is copied from https://github.com/abitofhelp/connect-go-example/blob/main/import.patch

It is required because of the issue described in https://github.com/bazelbuild/rules_go/issues/3951.
The connect protoc plugin generates files in a sub-directory, which is not supported by the rules_go.
This patch allows to override the import path of the generated files.

There is also an upstream discussion whether there should be support for generating files in
the directory itself rather than in a sub-directory. See:
https://github.com/connectrpc/connect-go/discussions/310

--- proto/def.bzl
+++ proto/def.bzl
@@ -123,8 +123,12 @@ def _go_proto_library_impl(ctx):
             imports = get_imports(ctx.attr, go.importpath),
             importpath = go.importpath,
         ))
+    importpath = None
+    if ctx.attr.overrideimportpath:
+        importpath = ctx.attr.overrideimportpath
     library = go.new_library(
         go,
+        importpath = importpath,
         resolver = _proto_library_to_source,
         srcs = go_srcs,
     )
@@ -164,6 +168,7 @@ go_proto_library = rule(
         "importpath": attr.string(),
         "importmap": attr.string(),
         "importpath_aliases": attr.string_list(),  # experimental, undocumented
+        "overrideimportpath": attr.string(),
         "embed": attr.label_list(providers = [GoLibrary]),
         "gc_goopts": attr.string_list(),
         "compiler": attr.label(providers = [GoProtoCompiler]),
-- 
